xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp

```diff
@@ -824,35 +872,49 @@ void CGUIDialogPVRChannelManager::SaveList()
   for (int iListPtr = 0; iListPtr < m_channelItems->Size(); ++iListPtr)
   {
     CFileItemPtr pItem = m_channelItems->Get(iListPtr);
+    if (pItem && pItem->GetProperty("Changed").asBoolean())
+    {
+      if (pItem->HasPVRChannelInfoTag())
+      {
+        if (pItem->GetProperty("SupportsSettings").asBoolean())
+          RenameChannel(pItem);
 
-    if (!pItem->HasPVRChannelInfoTag())
-      continue;
-
-    if (pItem->GetProperty("SupportsSettings").asBoolean())
-      RenameChannel(pItem);
+        PersistChannel(pItem, group, &iNextChannelNumber);
```

Comment:
```c++
if (PersistChannel(...))
  pItem->SetProperty("Changed", false);
```

to reset the flag only in case persist succeeds?

Reply:
No problem, question before I change this one though ... the only way it seems PersistChannel (and subsequently CPVRChannelGroup::UpdateChannel) can return false is if the channel item or group is no good, do you still see value in not resetting the flag given this?  With this change, I would like also add a subsequent check to make sure all the channel items persisted before re-enabling the Apply button, it might be awkward without an error message of some kind for the button to just stay enabled.

Reply:
I would  not over-engineer this. Just leave it as it is now with your PR.